commitlint.config.ts

```diff
@@ -207,6 +207,28 @@ module.exports = {
                         `Please watch out for leading or ending trailing whitespace`
                     ];
                 },
+
+                'type-space-before-paren': ({header}: {header:any}) => {
+                    let headerStr = convertAnyToString(header, "header");
+
+                    let offence = false;
+
+                    let colonIndex = headerStr.indexOf(":");
+                    if (colonIndex >= 0){
+                        let areaOrScope = headerStr.substring(0, colonIndex);
+                        let parenIndex = (areaOrScope.indexOf('('));
+                        if (parenIndex >= 1){
+                            if (headerStr[parenIndex - 1] === ' ') {
+                                offence = true;
+                            }
+                        }    
+                    }
+
+                    return [
+                        !offence,
+                        `Please watch out for leading or ending trailing whitespace`
```

Comment:
@realmarv well done so far! but we made a mistake here ^ (copy+paste problem hehehe)

Reply:
Done!